### PR TITLE
Add folium 0.14.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://github.com/python-visualization/folium/blob/v0.14.0/CHANGES.txt
License: https://github.com/python-visualization/folium/blob/v0.14.0/LICENSE.txt
Requirements:
- https://github.com/python-visualization/folium/blob/v0.14.0/pyproject.toml
- https://github.com/python-visualization/folium/blob/v0.14.0/requirements.txt
- https://github.com/python-visualization/folium/blob/a457a868ff17c109df170afa9c08b6c179b29d10/setup.py#L61

Actions:

- Remove `noarch`

- Skip `py<35`

- Add `--no-deps` to `script`

- Add missing `setuptools`, `setuptools_scm`, `wheel` to `host`

- Fix `python` in `host` and `run`

- Update `branca` pinning

- Add `license_family`

- Add `description`

- Add dev and doc urls

**_Notes_**

- folium depends on `branca` https://github.com/AnacondaRecipes/branca-feedstock/pull/3

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,44 +1,60 @@
...
+    - folium
   requires:
     - pip
   commands:
     - pip check
-  imports:
-    - folium
 
 about:
   home: https://github.com/python-visualization/folium
```
Comment (marcoesters) [inline recipe/meta.yaml]: The "official" home page appears to be https://python-visualization.github.io/folium/

**Review by marcoesters — COMMENTED** [review]
Do you know what's going on with the ibmz vs linux-s390x build. It seems odd that one machine can find branca 0.6.0 and the other one can't. This should be resolved before merging so that we can guarantee that the artifacts will be available on zeus.

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Thanks. Fixed

**Comment by skupr-anaconda** [comment]
> Do you know what's going on with the ibmz vs linux-s390x build. It seems odd that one machine can find branca 0.6.0 and the other one can't. This should be resolved before merging so that we can guarantee that the artifacts will be available on zeus.

Prefect has duplicate agents for linux-s390x (linux-s390x, ibmz, ibmz2) but only **one** of them will create artifacts. I hope this will be fixed soon

**Comment by skupr-anaconda** [comment]
`osx-64` and `osx-arm64` agents on `Prefect` failed:
`ssh: connect to host github.com port 22: Operation time out`

**Review by marcoesters — APPROVED** [review]
LGTM

**Pull request merged**